packages/dumb/src/components/trends/cbg/cbg-median-animated.tsx

```diff
@@ -0,0 +1,126 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import cx from 'classnames'
+import React, { FunctionComponent } from 'react'
+import { spring, TransitionMotion } from '@serprex/react-motion'
+
+import styles from './cbg-median-animated.css'
+import { getBgClass } from '../../../utils/blood-glucose/blood-glucose.util'
+import { BgBounds, ClassificationType } from '../../../models/blood-glucose.model'
+import { springConfig } from '../../../models/constants/animation.constants'
+
+interface TransitionMotionStyle {
+  height: number
+  median: number
+  opacity: number
+}
+
+interface TransitionMotionInterpolate {
+  key: string
+  style: TransitionMotionStyle
+}
+
+interface CBGMedianAnimatedProps {
+  bgBounds: BgBounds
+  datum: {
+    median: number
+    msX: number
+  }
+  defaultY: number
+  showingCbgDateTraces: boolean
+  sliceWidth: number
+  xScale: Function
+  yScale: Function
+}
+
+const DEFAULT_MEDIAN = 16
+
+export const CBGMedianAnimated: FunctionComponent<CBGMedianAnimatedProps> = (props) => {
+  const {
+    bgBounds,
+    datum,
+    showingCbgDateTraces,
+    sliceWidth,
+    xScale,
+    yScale
+  } = props
+
+  // default Y position is the center of the target range
+  // i.e., 100 mg/dL if target range is 80-120 mg/dL
+  const strokeWidth = sliceWidth / 8
+  const medianWidth = sliceWidth - strokeWidth
+  const width = medianWidth - strokeWidth
+  const medianHeight = medianWidth * 0.75
+  const x = xScale(datum.msX) - medianWidth / 2 + strokeWidth / 2
+  const defaultY = yScale(bgBounds.targetUpperBound - (bgBounds.targetUpperBound - bgBounds.targetLowerBound) / 2)
+
+  const defaultStyles = [{
+    key: 'median',
+    style: {
+      height: 0,
+      median: DEFAULT_MEDIAN,
+      opacity: 0
+    }
+  }]
+
+  const transitionMotionStyles = [{
+    key: 'median',
+    style: {
+      height: spring(medianHeight, springConfig),
+      median: spring(yScale(datum.median) - medianHeight / 2, springConfig),
+      opacity: spring(1.0, springConfig)
+    }
+  }]
+
+  const medianClasses = cx({
+    [styles.median]: true,
+    [styles[`${getBgClass(bgBounds, datum.median, ClassificationType.FiveWay)}FadeIn`]]: !showingCbgDateTraces,
+    [styles[`${getBgClass(bgBounds, datum.median, ClassificationType.FiveWay)}FadeOut`]]: showingCbgDateTraces
+  })
+
+  return (
+    <TransitionMotion
+      defaultY={defaultY}
+      defaultStyles={defaultStyles}
+      styles={transitionMotionStyles}
+    >
+      {(interpolatedStyles: TransitionMotionInterpolate[]) =>
+        interpolatedStyles.length !== 0 &&
+        <rect
+          className={medianClasses}
+          id={`cbgMedian-${interpolatedStyles[0].key}`}
+          data-testid={`cbgMedian-${interpolatedStyles[0].key}`}
```

Comment:
It might be worth defining `interpolatedStyles[0]` in a const given that it's the only value used from `interpolatedStyles`